### PR TITLE
set Minimum GITHUB_TOKEN permissions to github workflow

### PR DESCRIPTION
**Pull request by zhzhuang-zju**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
set Minimum GITHUB_TOKEN permissions to github workflow

**Which issue(s) this PR fixes**:
Parts of #5048 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -41,6 +45,8 @@ jobs:
           _output/release/${{ matrix.target }}-${{ matrix.os }}-${{ matrix.arch }}.tgz
           _output/release/${{ matrix.target }}-${{ matrix.os }}-${{ matrix.arch }}.tgz.sha256
   release-crds-assests:
+    permissions:
+      contents: write  # for softprops/action-gh-release to create GitHub release
```
Comment (liangyuanpeng) [inline .github/workflows/release.yml:49]: following the doc,set the correct permission.  https://github.com/softprops/action-gh-release?tab=readme-ov-file#permissions
/lgtm

**Comment by liangyuanpeng** [comment]
leave approval to @RainbowMango

**Review by RainbowMango — APPROVED** [review]
/approve

It makes sense that release action requires write permission.

**Pull request merged**